code/datums/gamemode/role/ninja.dm

```diff
@@ -285,6 +286,33 @@
 
 #define MAKE_SHURIKEN_COST 1000
 #define CHARGE_COST_MULTIPLIER 4
+
+/datum/action/item_action/make_shuriken
+	name = "Make Shuriken"
+	desc = "Fabricate a new shuriken."
+	icon_icon = 'icons/obj/weapons.dmi'
+	button_icon_state = "shuriken"
+
+/datum/action/item_action/make_shuriken/Trigger()
+	if (!owner.mind.GetRole(NINJA))
+		to_chat(owner, "<span class='warning'>Only a true ninja can do that!</span>")
```

Comment:
You lack HONOR to do that!

Reply:
what if it's a really honorable non-ninja, though
something something spider clan something something technowizardry